go/hyperkit.go

```diff
@@ -439,14 +443,43 @@ func (h *HyperKit) execHyperKit() error {
 	cmd := exec.Command(h.HyperKit, h.Arguments...)
 	cmd.Env = os.Environ()
 
-	// Plumb in stdin/stdout/stderr. If ConsoleStdio is configured
-	// plumb them to the system streams. If a logger is specified,
-	// use it for stdout/stderr logging. Otherwise use the default
-	// /dev/null.
+	// Plumb in stdin/stdout/stderr.
+	//
+	// If ConsoleStdio is configured and we are on a terminal,
+	// just plugin stdio. If we are not on a terminal we have a
+	// StateDir (as per checks above) and have configured HyperKit
+	// to use a PTY in the statedir. In this case, we just open
+	// the PTY slave and copy it to stdout (and ignore
+	// stdin). This allows for redirecting of the VM output, used
+	// for testing.
+	//
+	// If a logger is specified, use it for stdout/stderr
+	// logging. Otherwise use the default /dev/null.
 	if h.Console == ConsoleStdio {
-		cmd.Stdin = os.Stdin
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
+		if isTerminal(os.Stdout) {
+			cmd.Stdin = os.Stdin
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+		} else {
+			go func() {
+				ttyPath := fmt.Sprintf("%s/tty", h.StateDir)
+				var tty *os.File
+				var err error
+				for {
+					tty, err = os.OpenFile(ttyPath, os.O_RDWR, 0)
+					if err != nil {
```

Comment:
I'd have to plumb through a error channel and have `Start()` return an error. None of the function in the wrapper fail hard.